src/Illuminate/Support/Str.php

```diff
@@ -1097,7 +1099,11 @@ public static function random($length = 16)
                 $string .= substr(str_replace(['/', '+', '='], '', base64_encode($bytes)), 0, $size);
             }
 
-            return $string;
+            return match (true) {
+                $lower === true => static::lower($string),
+                $upper === true => static::upper($string),
+                default => $string,
+            };
```

Comment:
Talking of `Stringable`, maybe, your implementation would also benefit from it in combination with the `Conditional` trait